model/lifecycle.go

```diff
@@ -67,20 +67,38 @@ func SetVersionActivation(versionId string, active bool, caller string) error {
 		build.ByVersion(versionId).WithFields(build.IdKey),
 	)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "can't get builds for version '%s'", versionId)
 	}
-	for _, b := range builds {
-		err = SetBuildActivation(b.Id, active, caller, false)
-		if err != nil {
-			return err
-		}
+	buildIDs := make([]string, 0, len(builds))
+	for _, build := range builds {
+		buildIDs = append(buildIDs, build.Id)
 	}
-	return nil
+
+	if err = build.UpdateActivation(buildIDs, active, caller); err != nil {
+		return errors.Wrapf(err, "can't set activation for builds in '%s'", versionId)
+	}
+
+	catcher := grip.NewBasicCatcher()
+	for _, buildID := range buildIDs {
+		catcher.Add(SetTaskActivationForBuild(buildID, active, caller, false))
```

Comment:
Previously deactivating tasks and builds both happened inside `SetBuildActivation`. To deactivate a version we would call `SetBuildActivation` in a loop on all the builds in the version. Since deactivating many tasks took a while deactivation of each build was separated by many seconds.

The refactor is to break up deactivating the builds and tasks into separate functions so we can deactivate all the builds at the same time, so there's no intermediate time where some builds are active and some are inactive.

Reply:
could this become some kind of updateMany?

Reply:
done.